### PR TITLE
Allow max variable length when shifting

### PR DESCRIPTION
**Pull request by stefanahman**
Fixes issue with segments that are full lengths and contains a separator.

TP: https://apoexab.tpondemand.com/entity/14249

### CODE REVIEW

File: spec/gs1/barcode/healthcare_spec.rb [hunk 1]
```diff
@@ -27,6 +27,17 @@
         expect(subject.serial_number).to eq(GS1::SerialNumber.new('123123'))
       end
     end
+
+    context "01070462606504112101582353425597816838\u001E1721093010184238131" do
```
Comment (joakimlagergren) [inline spec/gs1/barcode/healthcare_spec.rb]: Perhaps a name that is more descriptive of the case being tested?

Comment (stefanahman) [reply spec/gs1/barcode/healthcare_spec.rb]: Output:
image

How would you name this?

Comment (stefanahman) [reply spec/gs1/barcode/healthcare_spec.rb]: Result:
image

File: spec/gs1/barcode/healthcare_spec.rb [hunk 2]
```diff
@@ -27,6 +27,17 @@
         expect(subject.serial_number).to eq(GS1::SerialNumber.new('123123'))
       end
     end
+
+    context "maximum length of serial number + separator" do
```
Comment (joakimlagergren) [inline spec/gs1/barcode/healthcare_spec.rb]: 👍

**Pull request merged**